src/main/kotlin/no/nav/syfo/oppfolgingstilfelle/bit/Oppfolgingstilfellebit.kt

```diff
@@ -0,0 +1,154 @@
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and Tag.KORRIGERT_ARBEIDSTID and Tag.BEHANDLINGSDAGER,
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and Tag.KORRIGERT_ARBEIDSTID and Tag.FULL_AKTIVITET,
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and Tag.KORRIGERT_ARBEIDSTID and (Tag.GRADERT_AKTIVITET or Tag.INGEN_AKTIVITET),
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and (Tag.PERMISJON or Tag.FERIE),
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and (Tag.EGENMELDING or Tag.PAPIRSYKMELDING or Tag.FRAVAR_FOR_SYKMELDING),
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and ListContainsPredicate.tagsSize(2),
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and Tag.BEHANDLINGSDAG,
+            Tag.SYKEPENGESOKNAD and Tag.SENDT and Tag.BEHANDLINGSDAGER,
+            Tag.SYKMELDING and (Tag.SENDT or Tag.BEKREFTET) and Tag.PERIODE and Tag.BEHANDLINGSDAGER,
+            Tag.SYKMELDING and (Tag.SENDT or Tag.BEKREFTET) and Tag.PERIODE and Tag.FULL_AKTIVITET,
+            Tag.SYKMELDING and (Tag.SENDT or Tag.BEKREFTET) and Tag.PERIODE and (Tag.GRADERT_AKTIVITET or Tag.INGEN_AKTIVITET),
+            Tag.SYKMELDING and Tag.BEKREFTET and Tag.ANNET_FRAVAR,
+            Tag.SYKMELDING and Tag.SENDT and Tag.PERIODE and Tag.REISETILSKUDD and Tag.UKJENT_AKTIVITET,
+            Tag.SYKMELDING and Tag.NY and Tag.PERIODE and Tag.BEHANDLINGSDAGER,
+            Tag.SYKMELDING and Tag.NY and Tag.PERIODE and Tag.FULL_AKTIVITET,
+            Tag.SYKMELDING and Tag.NY and Tag.PERIODE and (Tag.GRADERT_AKTIVITET or Tag.INGEN_AKTIVITET),
+            Tag.SYKMELDING and Tag.NY and Tag.PERIODE and Tag.REISETILSKUDD and Tag.UKJENT_AKTIVITET,
+        )
+    }
+}
+
+fun List<OppfolgingstilfelleBit>.toOppfolgingstilfelleDagList(): List<OppfolgingstilfelleDag> {
+    require(this.isNotEmpty())
+
+    val firstFom = this.firstFom().toLocalDate()
+    val lastTom = this.lastTom().toLocalDate()
+
+    return (0..ChronoUnit.DAYS.between(firstFom, lastTom))
+        .map { dayInt ->
+            this.pickOppfolgingstilfelleDag(
+                dag = firstFom.plusDays(dayInt),
+            )
+        }
+}
+
+fun List<OppfolgingstilfelleBit>.bit(
+    comparator: Comparator<OppfolgingstilfelleBit>,
+): OppfolgingstilfelleBit = this.sortedWith(comparator).first()
+
+fun List<OppfolgingstilfelleBit>.firstFom(): LocalDateTime =
+    this.bit(
+        comparator = Comparator.comparing {
+            it.fom
+        }
+    ).fom
+
+fun List<OppfolgingstilfelleBit>.lastTom(): LocalDateTime =
+    this.bit(
+        comparator = Comparator.comparing<OppfolgingstilfelleBit, LocalDateTime> {
+            it.tom
+        }.reversed()
+    ).tom
+
+fun List<OppfolgingstilfelleBit>.pickOppfolgingstilfelleDag(
+    dag: LocalDate,
+): OppfolgingstilfelleDag {
+    val bitList = this.filter {
+        dag in (it.fom.toLocalDate()..(it.tom.toLocalDate()))
+    }
+    return bitList
+        .groupBy { bit -> bit.inntruffet.toLocalDate() }
```

Comment:
vil inntruffet alltid være mellom fom og tom?

Reply:
Det tror jeg ikke.

Reply:
Det underliggende spørsmålet blir i alle fall om det feks hadde vært riktigere å gruppere på fom?
(jeg sier ikke at det er det 😉, men jeg tror argumentene for at det er inntruffet som er riktig her forklarer noe vesentlig om modellen).

Reply:
Slik koden er nå så vil det som skjedde sist(mottatt sykmelding eller sykepengesøknad) blir det som blir valgt som gjeldende for en gitt dag, og dersom det er motstridende innpå blant biter innenfor denne grupperingen så vil den med høyest presendens få prioritet.
Gitt at man ønsker å ha en slik gruppering, så tenker jeg at å gruppere på f.eks fom tenker jeg er mindre treffende enn å gruppere på intruffetDato.